cookbook/ifan02.rst

```diff
@@ -0,0 +1,181 @@
+Ifan02
+======
+
+.. seo::
+    :description: Instructions for using Sonoff ifan02 in ESPHome.
+    :keywords: Fan, Sonoff, ifan02
+
+
+Configuration for `Sonoff ifan02 <https://www.itead.cc/sonoff-ifan02-wifi-smart-ceiling-fan-with-light.html>`__.
+
+First you need a :doc:`components/output/custom.html` to control the ifan02.
+
+Create a ifan02.h file:
+
+.. code-block:: c++
+
+    #include "esphome.h"
+    using namespace esphome;
+
+    class IFan02Output : public Component, public output::FloatOutput {
+      public:
+        void write_state(float state) override {
+            if (state < 0.3) {
+              digitalWrite(5, LOW);
+              digitalWrite(4, LOW);
+              digitalWrite(15, LOW);
+            }
+
+            if (state >= 0.32 && state <= 0.34) {
+              digitalWrite(5, HIGH);
+              digitalWrite(4, LOW);
+              digitalWrite(15, LOW);
+            }
+            if (state >= 0.65 && state <= 0.67) {
+              digitalWrite(5, HIGH);
+              digitalWrite(4, HIGH);
+              digitalWrite(15, LOW);
+            }
+            if (state >= 0.9) {
+              digitalWrite(5, HIGH);
+              digitalWrite(4, LOW);
+              digitalWrite(15, HIGH);
+            }
+        }
```

Comment:
rewrite this to be
```cpp
if (state < 0.3) {
  // OFF
} else if (state < 0.6) {
  // low speed
} else if (state < 0.9) {
  // medium speed
} else {
  // high speed
}
```

There are some values that this component does not handle like 0.31 (the if/else if fixes that kinda)